src/RecordSession.cc

```diff
@@ -1052,8 +1059,8 @@ static bool inject_handled_signal(RecordTask* t) {
     // kill the process after this. Stash the signal and make sure
     // we know to treat it as fatal when we inject it. Also disable the
     // signal handler to match what the kernel does.
-    t->set_sig_blocked(sig, false);
-    t->set_sig_handler_default(sig);
+    t->set_sig_blocked(SIGSEGV, false);
+    t->set_sig_handler_default(SIGSEGV);
```

Comment:
The kernel only does this for SIGSEGV.  See https://github.com/torvalds/linux/blob/2b1ecc3d1a6b10f8fbac7f83d80db30b5a2c2791/kernel/signal.c#L1446

`clone_bad_stack` triggers this.  Setting up the signal handler frame for `SYSCALLBUF_DESCHED_SIGNAL` fails because the stack is bad.  That triggers a SIGSEGV which puts us in this block of code here.  If we default away `SYSCALLBUF_DESCHED_SIGNAL` then `verify_signal_state` asserts.